app/jobs/full_text_to_solr_job.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+##
+# a job class that will update a Solr document
+# with the extracted text of an attached PDF
+class FullTextToSolrJob < ApplicationJob
+  # @return [String] the id for the SolrDocument being updated
+  # @return [IO] the PDF content
+  attr_reader :solr_document_id, :work_content
+
+  queue_as :default
+
+  def initialize(solr_document_id, work_content)
+    @solr_document_id = solr_document_id
+    @work_content = work_content
+  end
+
+  def perform
+    solr = RSolr.connect(url: ActiveFedora::SolrService.instance.conn.uri.to_s)
+    solr_document = SolrDocument.find(@solr_document_id).to_h
+
+    # Append additional text if there is already text in this field
+    if solr_document["all_text_timv"]
```

Comment:
@little9 What happens if this object is being re-indexed? Will copies of the same text keep accumulating?

Reply:
Oh, that's true.

Reply:
@bess Yeah, my thinking behind that was that there may be Works that that multiple files so you would need to run the operation for each of the files to capture all of the text. It might not be necessary if in Alexandria, there is one PDF for the Work in the collections that this will be used on. If that's the case I'll change this to not append additional text. @dunn do you have thoughts about this?

Reply:
ETDs may come with arbitrary supplemental files, including more PDFs.  So there might be multiple sources needing to be indexed to that field (@chrissyrissmeyer unless you think supplemental PDFs shouldn't be indexed).  But that still leaves the problem of re-indexing; does a re-index start from scratch, or modify the existing Solr document?

Reply:
I would say that it is modifying the existing document. It grabs the existing document, appends to (if that field is present) or creates the `all_text_timv` field and then commits that modified document.

Reply:
OK, then we need to figure out how to avoid duplicate text when we re-index.

Reply:
@bess do you think it would be OK for me to remove the appending of the text. This would mean each time it is run it will be replaced and you would be getting the full text from the one attached file that run this on. Then we could add another story for adding full text for all the PDFs in the work.

Reply:
If there are multiple PDFs, how does it decide which one to run this one?

Reply:
It'll be the one that you pick when running it:

```ruby 
FullTextToSolrJob.new("fk4jq25q2f", Image.first.file_sets[0].files.first.content).perform
```